### PR TITLE
fix AppBar docs for backgroundColor & foregroundColor

### PR DESCRIPTION
**Pull request by BiskupMaik**
Before the M3 migration, the doc made sense. but after, it needed to specify which material version does this issue occur in.
The _AppBarDefaultsM3 don't take the brightness in consideration anymore:

image

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
Thanks for clarifying this!

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -514,10 +518,13 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
...
-  /// [ColorScheme.onPrimary] if the overall theme's brightness is
-  /// [Brightness.light], and [ColorScheme.onSurface] if the overall
-  /// theme's brightness is [Brightness.dark].
+  /// value is also null:
+  /// In Material v2 (i.e., when [ThemeData.useMaterial3] is false),
+  /// then [AppBar] uses the overall theme's [ColorScheme.onPrimary] if the
+  /// overall theme's brightness is [Brightness.light], and [ColorScheme.onSurface]
+  /// if the overall theme's brightness is [Brightness.dark].
+  /// In Material v3 (i.e., when [ThemeData.useMaterial3] is true),
+  /// then [AppBar] uses the overall theme's [ColorScheme.onSurface]
```
Comment (cbracken) [inline packages/flutter/lib/src/material/app_bar.dart:527]: nit: add a trailing period.

**Review by cbracken — APPROVED** [review]
lgtm with one small fix!

File: packages/flutter/lib/src/material/app_bar.dart (hunk 1 above)
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart:527]:
```suggestion
  /// then [AppBar] uses the overall theme's [ColorScheme.onSurface].
```

**Review by HansMuller — APPROVED** [review]
Added the period.

**Pull request merged**